ncs/ncs_scraper.py

```diff
@@ -0,0 +1,218 @@
+        urls[state.get_text()] = state.find_parent('a')['href']
+
+    with open('state_links.json','w') as f:
+        f.write(json.dumps(urls, indent = 4, sort_keys = True))
+
+    return urls
+
+def get_job_urls(soup):
+
+    all_job_urls = soup.findAll('a', attrs={'onclick': re.compile('ViewJobPopup\(.*\)')})
+
+    all_job_urls_list = []
+
+    for element in all_job_urls:
+
+        job_url = element['onclick']
+
+        all_job_urls_list.append(re.search('.*(https\:.+)\'', job_url).group(1))
+
+    return list(set(all_job_urls_list))
+
+
+def get_landing_page_job_details(soup):
+    """
+    :param soup:
+    :return: list of dictionaries
+
+   {
+	'Company:':
+	'Job Location:':
+	'Salary:':
+	'Skill Required:':
+	'Job Description:':
+	}
+
+    """
+
+    job_details_list = []
+
+
+    for tab_soup in soup.findAll('div', attrs={'class' : re.compile('row padding0-15'), 'id': 'mytab'}):
+
+        job_details = dict()
+
+
+        try:
+
+            # Job URL
+            job_url = tab_soup.find('a', attrs={'onclick': re.compile('ViewJobPopup\(.*\)')})
+            job_url = job_url['onclick']
+            job_details['job_url'] = re.search('.*(https\:.+)\'', job_url).group(1)
+
+            job_details['posted_on'] = tab_soup.find_all('span', attrs={'class': 'text-muted pull-right'})[0].find_next().get_text()
+
+
+            # Job Details from landing page
+            text_info = tab_soup.find_all('span', attrs={'class': 'text-info'})
+
+        except Exception:
+            print("ERROR")
+
+
+        for txt in text_info:
+            job_details[clean_text(txt.get_text())] = clean_text(txt.find_next().get_text())
+
+        job_details_list.append(job_details)
+
+
+    return job_details_list
+
+
+def run_process():
+    state_urls = get_state_urls()
+
+    for state, state_url in state_urls.items():
+
+
+        print("starting scraping for state {}".format(state))
+
+        # State landing Page
+        soup = scraper.get_soup(state_url)
+
+
+        job_details_list = get_landing_page_job_details(soup)
+
+        
+        driver = webdriver.Chrome()
+        driver.get(state_url)
+        
+        #FINDING: total number of pages 
+        #the total pages in written in the form of text below the page buttons list
+        getelementbyxpath=driver.find_element_by_xpath('/html/body/form/div[4]/div/div[4]/div[1]/div/span/div[1]/div/div/div[1]/div/div[2]/div/div/div[1]/div[3]/div/div[1]/div/div[2]/nav[1]/div[2]')
+        s=getelementbyxpath.find_elements_by_xpath(".//*")[0].text
+        number_of_pages=[int(i) for i in s.split() if i.isdigit()][0]
+
+        
+        # Pagination 
+        for page in range(number_of_pages):
+        
+            print(page)
+            if(page>0):
+                
+#                 getelementbyxpath=driver.find_element_by_xpath('/html/body/form/div[4]/div/div[4]/div[1]/div/span/div[1]/div/div/div[1]/div/div[2]/div/div/div[1]/div[3]/div/div[1]/div/div[2]/nav[1]/div[1]')
```

Comment:
If commented code is being kept for future purposes, please explicitly add a comment stating the reason. Otherwise remove it